File: demo/src/main/java/com/bytedance/scenedemo/async_inflate/AsyncInflateSceneDemo.java
package com.bytedance.scenedemo.async_inflate;

import android.graphics.drawable.AnimatedVectorDrawable;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.ImageView;

import com.bytedance.scene.group.AsyncLayoutGroupScene;
import com.bytedance.scene.group.GroupScene;
import com.bytedance.scenedemo.R;

/**
 * Created by JiangQi on 9/19/18.
 */
public class AsyncInflateSceneDemo extends GroupScene {

    private int id;

    @NonNull
    @Override
    public ViewGroup onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        id = View.generateViewId();

        FrameLayout frameLayout = new FrameLayout(requireActivity());
        frameLayout.setId(id);
        return frameLayout;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        TestAsyncInflateScene scene = new TestAsyncInflateScene();
        scene.setAsyncLayoutEnabled(true);

        add(id, scene, "wo");
    }


    public static class TestAsyncInflateScene extends AsyncLayoutGroupScene {
        @Override
        protected int getLayoutId() {
            return R.layout.layout_test_async_layout;
        }

        @Override
        public void onAsyncActivityCreated(Bundle savedInstanceState) {
            super.onAsyncActivityCreated(savedInstanceState);

            ((AnimatedVectorDrawable) ((ImageView) findViewById(R.id.imageview)).getDrawable()).start();
        }
    }
}

File: demo/src/main/java/com/bytedance/scenedemo/group/async_inflate/AsyncInflateSceneDemo.kt
package com.bytedance.scenedemo.group.async_inflate

import com.bytedance.scene.group.GroupScene
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import android.widget.FrameLayout
import com.bytedance.scenedemo.group.async_inflate.AsyncInflateSceneDemo.TestAsyncInflateScene
import com.bytedance.scene.group.AsyncLayoutGroupScene
import com.bytedance.scenedemo.R
import android.graphics.drawable.AnimatedVectorDrawable
import android.view.View
import android.widget.ImageView

/**
 * Created by JiangQi on 9/19/18.
 */
class AsyncInflateSceneDemo : GroupScene() {
    private var id = 0
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup, savedInstanceState: Bundle?): ViewGroup {
        id = View.generateViewId()
        val frameLayout = FrameLayout(requireActivity())
        frameLayout.id = id
        return frameLayout
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        val scene = TestAsyncInflateScene()
        scene.isAsyncLayoutEnabled = true
        add(id, scene, "wo")
    }

    class TestAsyncInflateScene : AsyncLayoutGroupScene() {
        override fun getLayoutId(): Int {
            return R.layout.layout_test_async_layout
        }

        override fun onAsyncActivityCreated(savedInstanceState: Bundle?) {
            super.onAsyncActivityCreated(savedInstanceState)
            ((findViewById<View>(R.id.imageview) as ImageView?)!!.drawable as AnimatedVectorDrawable).start()
        }
    }
}